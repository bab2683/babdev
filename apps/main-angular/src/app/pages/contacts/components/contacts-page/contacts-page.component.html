<ng-container *ngIf="pageData$ | async as pageData">
  <babdev-page-with-bg
    [bgData]="pageData && pageData.bgData"
    [title]="'contacts.TITLE' | translate | async"
  >
    <ul class="contact-links">
      <li
        class="contact-links__item contact-links__item--{{ link.id }}"
        *ngFor="let link of links"
        [ngSwitch]="link.type"
      >
        <fa-icon
          class="contact-links__icon"
          *ngIf="link.icon"
          [icon]="link.icon"
        ></fa-icon>
        <div class="contact-links__text-container">
          <span
            class="contact-links__text"
            babdevClickable
            [clickableActions]="link.actions"
            >{{ link.text }}</span
          >
        </div>
      </li>
    </ul>
  </babdev-page-with-bg>
</ng-container>
