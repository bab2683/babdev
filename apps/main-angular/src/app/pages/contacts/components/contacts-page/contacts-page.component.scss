@import "~@styleguide_funcional";

$prefix: ".contact-links";

#{$prefix} {
  margin-top: px-to-rem(40);

  &__icon {
    font-size: px-to-rem(60);
  }

  &__text-container {
    flex: 1;
    text-align: center;
  }

  &__text {
    color: transparent;

    @include transition("color");
  }

  &__item {
    align-items: center;
    display: flex;
    padding: px-to-rem(5);
    position: relative;

    @include transition(("color", "background-color"));

    &::before {
      bottom: 0;
      content: "";
      left: 0;
      position: absolute;
      top: 0;
      transform: scaleX(0);
      transform-origin: left;
      width: 100%;
      z-index: -1;

      @include transition("transform");
    }

    @each $name, $color in $brand-colors {
      &--#{$name} {
        /* stylelint-disable */
        &::before {
          background-color: map-get($map: $brand-color-shades, $key: "#{$name}-light");
        }

        &:hover {
          color: $color;

          #{$prefix}__text {
            color: $color;
          }

          &::before {
            transform: scaleX(1);
          }
        }
        /* stylelint-enable */
      }
    }
  }
}
