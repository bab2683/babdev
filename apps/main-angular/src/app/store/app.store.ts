import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { ActionReducerMap, MetaReducer } from '@ngrx/store';

import { environment } from '@env/environment';
import {
  GlobalEffects,
  globalReducer,
  GlobalState,
  GLOBAL_FEATURE,
  ROUTER_FEATURE
} from './features';

export interface AppState {
  [GLOBAL_FEATURE]: GlobalState;
  [ROUTER_FEATURE]: RouterReducerState;
}

export const AppEffects: any[] = [GlobalEffects];

export const reducers: ActionReducerMap<AppState> = {
  [GLOBAL_FEATURE]: globalReducer,
  [ROUTER_FEATURE]: routerReducer
};

export const metaReducers: Array<MetaReducer<
  AppState
>> = !environment.production ? [] : [];
