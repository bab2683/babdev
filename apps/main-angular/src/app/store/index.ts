export * from './app.store';
export * from './features';
export * from './mocks';
