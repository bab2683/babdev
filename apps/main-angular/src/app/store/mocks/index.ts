export * from './app.store.mock';
