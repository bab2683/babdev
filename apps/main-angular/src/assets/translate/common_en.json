{
  "PAGES": {
    "HOME": "Home",
    "CONTACTS": "Contacts"
  },
  "TIME": {
    "YEAR": {
      "SINGLE": "year",
      "MULTIPLE": "years"
    },
    "MONTH": {
      "SINGLE": "month",
      "MULTIPLE": "months"
    },
    "DAY": {
      "SINGLE": "day",
      "MULTIPLE": "days"
    }
  }
}
