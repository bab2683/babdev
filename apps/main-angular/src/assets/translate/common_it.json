{
  "PAGES": {
    "HOME": "Home",
    "CONTACTS": "Contatti"
  },
  "TIME": {
    "YEAR": {
      "SINGLE": "anno",
      "MULTIPLE": "anni"
    },
    "MONTH": {
      "SINGLE": "mese",
      "MULTIPLE": "mesi"
    },
    "DAY": {
      "SINGLE": "giorno",
      "MULTIPLE": "giorni"
    }
  }
}
