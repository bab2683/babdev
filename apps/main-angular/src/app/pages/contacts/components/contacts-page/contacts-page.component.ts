import { Component, OnInit } from '@angular/core';
import { BackGroundPositionYEnum, ClickableActions } from '@babdev/layout';
import { DictionaryLoader, TranslateService } from '@babdev/translate';
import { select, Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import { ContactLinks } from '@constants';
import { ContactLinkTypeEnum } from '@enums';
import { ContactsLink } from '@models';
import { AppState, getIsMobileState } from '@store';

@Component({
  selector: 'babdev-contacts-page',
  templateUrl: './contacts-page.component.html',
  styleUrls: ['./contacts-page.component.scss']
})
export class ContactsPageComponent implements OnInit {
  public isMobile$: Observable<boolean>;

  public actions = ClickableActions;
  public links: ContactsLink[] = ContactLinks;
  public itemType = ContactLinkTypeEnum;
  public bgPath: string = 'bg/contacts';
  public bgPositionY: BackGroundPositionYEnum = BackGroundPositionYEnum.BOTTOM;

  private dictionary: DictionaryLoader = { location: '/pages/contacts/', name: 'contacts' };

  constructor(private translateService: TranslateService, private store: Store<AppState>) {}

  ngOnInit() {
    this.translateService.loadDictionary(this.dictionary);
    this.isMobile$ = this.store.pipe(select(getIsMobileState));
  }
}
