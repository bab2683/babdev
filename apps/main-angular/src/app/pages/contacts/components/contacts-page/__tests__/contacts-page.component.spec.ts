import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { TranslateService } from '@babdev/translate';
import { TranslatePipeMock, TranslateServiceMock } from '@babdev/translate-testing';
import { Store } from '@ngrx/store';
import { MockStore, provideMockStore } from '@ngrx/store/testing';

import { AppState, GLOBAL_FEATURE } from '@store';

import { ContactsPageComponent } from '../contacts-page.component';

describe('ContactsPageComponent', () => {
  let component: ContactsPageComponent;
  let fixture: ComponentFixture<ContactsPageComponent>;
  let store: MockStore<AppState>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ContactsPageComponent, TranslatePipeMock],
      providers: [
        {
          provide: TranslateService,
          useClass: TranslateServiceMock
        },
        provideMockStore({ initialState: { [GLOBAL_FEATURE]: { isMobile: false } } })
      ],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ContactsPageComponent);
    component = fixture.componentInstance;
    store = TestBed.get<Store<AppState>>(Store);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
