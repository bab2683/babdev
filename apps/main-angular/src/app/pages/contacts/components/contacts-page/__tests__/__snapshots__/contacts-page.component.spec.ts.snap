// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsPageComponent should match snapshot 1`] = `
<babdev-contacts-page
  actions={[Function Object]}
  bgPath={[Function String]}
  bgPositionY={[Function String]}
  dictionary={[Function Object]}
  isMobile$={[Function Store]}
  itemType={[Function Object]}
  links={[Function Array]}
  store={[Function MockStore]}
  translateService={[Function TranslateServiceMock]}
>
  <babdev-page-with-bg
    title="contacts.TITLE"
  >
    <ul
      class="contact-links"
    >
      
      <li
        class="contact-links__item contact-links__item--ln"
        ng-reflect-ng-switch="url"
      >
        
        <fa-icon
          class="contact-links__icon"
        />
        <div
          class="contact-links__text-container"
        >
          <span
            babdevclickable=""
            class="contact-links__text"
          >
            linkedin.com/in/bab2683-braulio-barahona/
          </span>
        </div>
      </li>
      <li
        class="contact-links__item contact-links__item--gh"
        ng-reflect-ng-switch="url"
      >
        
        <fa-icon
          class="contact-links__icon"
        />
        <div
          class="contact-links__text-container"
        >
          <span
            babdevclickable=""
            class="contact-links__text"
          >
            github.com/bab2683
          </span>
        </div>
      </li>
      <li
        class="contact-links__item contact-links__item--fb"
        ng-reflect-ng-switch="url"
      >
        
        <fa-icon
          class="contact-links__icon"
        />
        <div
          class="contact-links__text-container"
        >
          <span
            babdevclickable=""
            class="contact-links__text"
          >
            facebook.com/bab2683
          </span>
        </div>
      </li>
      <li
        class="contact-links__item contact-links__item--gm"
        ng-reflect-ng-switch="mail"
      >
        
        <fa-icon
          class="contact-links__icon"
        />
        <div
          class="contact-links__text-container"
        >
          <span
            babdevclickable=""
            class="contact-links__text"
          >
            barahona.braulio@gmail.com
          </span>
        </div>
      </li>
      <li
        class="contact-links__item contact-links__item--sk"
        ng-reflect-ng-switch="text"
      >
        
        <fa-icon
          class="contact-links__icon"
        />
        <div
          class="contact-links__text-container"
        >
          <span
            babdevclickable=""
            class="contact-links__text"
          >
            bab2683
          </span>
        </div>
      </li>
    </ul>
  </babdev-page-with-bg>
</babdev-contacts-page>
`;
